### DISCUSSION TITLE
[预览]v2.0.0

### DISCUSSION THREAD

**Discussion by the1812** [the1812/Bilibili-Evolved#2448] [分享 / Share]
🎃**欢迎来到 Bilibili Evolved v2.0.0 (预览版), 为了更长远的发展, 我完全重写了整个项目:**
- 引入现代化前端工具 (webpack, Babel, PostCSS 等)
- 源代码 TypeScript 全覆盖 (webpack 等配置文件除外)
- 自带 UI 组件库 (基于 Vue)
- 移除对 jQuery 和 Vuex 的依赖
- 全新的架构设计, 脚本体积大幅缩小: 2.79MB 👉 759KB
- 不再有离线版, 你可以自行控制功能的更新

**需要注意的是:**
- 脚本不会由脚本管理器自动更新到 v2, 更新提示里也不会直接让你安装更新. 因为 v2 的安装和使用方式和 v1 有很大区别, 即便你是 v1 的老用户, 也强烈建议你重新阅读 [README](https://github.com/the1812/Bilibili-Evolved/blob/preview/README.md).
- 安装前建议先备份 v1 设置, 然后卸载或者清空 v1 的数据存储, 避免遗留数据继承至 v2 造成性能影响. 

**功能差异:**
- 虽然 README 有写, 但还是要再强调下, 全新安装的 v2 不包含任何实质性的功能, 它本身更接近于一个功能管理器, 你可以装个 `v1 设置迁移` 导入 v1 的设置 (顺便还能熟悉下怎么装功能), v1 开着的功能会在 v2 中自动安装, 详情见[此文档](https://github.com/the1812/Bilibili-Evolved/blob/preview/doc/v1-migrate.md).
- 暂不支持 v1 的 `简化首页` 功能, 在 v2 中 `简化首页` 等于 v1 的 `首页过滤` 功能. v1 的 `简化首页` 由于代码设计过于糟糕, 很难再进一步扩充功能, 我计划在 v2 发布后重写这个功能, 届时会比 v1 更加美观和强大. 如果你非常需要这个功能, 请暂时不要升级到 v2.
- 暂不支持 Safari, 但也有计划去支持了. (#2349)
- 暂不支持 v1 的 `界面翻译` 功能, 以后可能会做英语支持.
- 下载视频暂不支持 ffmpeg 命令生成 和 课程下载.
- 下载视频删除了`直接下载`的功能 (遗留 bug 过多), 改为 Toast 输出, 区别是可以直接调用浏览器下载, 但不能给文件自动命名. (aria2 输出依然可以自动命名)

安装及使用方法见 [README](https://github.com/the1812/Bilibili-Evolved/blob/preview/README.md#安装). 如希望参与开发, 请参考[代码贡献指南](https://github.com/the1812/Bilibili-Evolved/blob/preview/CONTRIBUTING.md).

如有其他疑问, 请在这个 Release 对应的讨论区里回复.

-----

**写给 Technical Preview 用户:**
v2 现在正式结束 TP 了, 可以导出设置, 删除 TP 版并更换 Preview 版, 功能已和 `v1.12.21` 同步. 更新源之类的设置换回 `jsDelivr` 应该也能正常使用了.
从 TP 版导出的设置里功能可能还是指向 `v2` 分支的, 要接收最新更新, 可以安装 `更新链接替换` 组件, 然后在功能面板里启动, 输入分支名 `preview` 就可以把已安装功能的更新链接分支切换过来. 弄完之后可以删掉 `更新链接替换`.
感谢参与 v2 的建设!

<hr /><em>This discussion was created from the release <a href='https://github.com/the1812/Bilibili-Evolved/releases/tag/v2.0.0-preview'>[预览]v2.0.0</a>.</em>

-----

**后续更新传送**
- #2495 
- #2526
- #2571
- #2575
- #2592
- #2607

**Comment by ZiSaAcZ7** [comment]
好耶！但是我想用简化首页所以等那个更新了或者V1出重大bug我再换吧。顺便我为项目捐款了CNY198作为感谢

**Reply by the1812** [reply]
感谢支持!

**Comment by kethoven** [comment]
清完，如果组件更新，是不是只能够一个一个组件点击更新，不能够统一更新组件。

**Reply by the1812** [reply]
有个自动更新器组件可以统一更新

**Reply by kethoven** [reply]
> 有个自动更新器组件可以统一更新

谢谢！

**Comment by the1812** [comment]
不知道为啥通过最近几个 issue 发现, 那个`我全都要`合集包似乎有着很大诱惑力啊(
但是最好不要去装那个, 那个合集包只是当时做着玩的, 囊括所有功能就代表着总有些功能是你不想要的, 甚至是不明不白装了之后导致你不会用了的, 所以这个合集包我先删掉了.

**Reply by oscarwty** [reply]
毕竟大部分软件反而是选全部安装才能最大限度避免出问题（（（
以及那个右侧提示挡住的bug就拜托大佬了！

**Reply by the1812** [reply]
挡住其实再点一下在线仓库让设置面板收起就可以了

**Reply by oscarwty** [reply]
> 
> 
> 挡住其实再点一下在线仓库让设置面板收起就可以了

好的谢谢！

**Reply by kethoven** [reply]
> 挡住其实再点一下在线仓库让设置面板收起就可以了



> 不知道为啥通过最近几个 issue 发现, 那个`我全都要`合集包似乎有着很大诱惑力啊( 但是最好不要去装那个, 那个合集包只是当时做着玩的, 囊括所有功能就代表着总有些功能是你不想要的, 甚至是不明不白装了之后导致你不会用了的, 所以这个合集包我先删掉了.

装完之后，再去自己删减，挺好的。
其他安装包，有些自己想要的功能，可能不全。
还是全部都安装后，再去卸载不需要的功能。

**Reply by the1812** [reply]
- 对于 v1 用户, 用 `v1设置迁移` 是最好的方式, 不会安装不需要的功能;
- 对于将来的新用户, 上来就是 v2, 他们必须得经历一次功能挑选, 搭配出想要使用的功能;

这两种都不会用到`我全都要`, 另外也有个现成的例子就是 VS Code, 它的插件也支持合集包, 但不会有这种安装所有功能的合集包.

**Comment by LoneFireBlossom** [comment]
我用广告屏蔽的插件搞的简化首页，十分简洁🤣（对知乎我也是这么干的）
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/50087596/140032972-1c729349-8ad1-4195-8deb-e87e7b914a6d.png">

**Comment by liangzilixueha** [comment]
完全重写我的天，感谢大佬的无私奉献

**Comment by BunnySmasher** [comment]
大佬反馈一个问题:更换勋章,不能滑动,自动排序
![QQ截图20211104093910](https://user-images.githubusercontent.com/34413382/140242955-a6dd56a2-f865-4b32-aa7d-de3ed1a75b5c.png)

**Reply by the1812** [reply]
头一次见这么多的... 你还需要排序吗, 按照名称还是等级排?

**Reply by BunnySmasher** [reply]
就跟你们1.0的版本一样就蛮好的,按照等级吧

**Reply by ChaXxl** [reply]
你这勋章也。。。太多了吧

**Comment by ciring** [comment]
![image](https://user-images.githubusercontent.com/33834625/140643672-a82afc24-7d91-4468-954a-e965919cfbf0.png)
勋章字体看不清 更换文本颜色还是看不清。
顺便问问V2的自动佩戴勋章的功能还在开发吗，只看到个选择勋章，没看到自动更换

**Reply by ciring** [reply]
补充：不知道是不是因为我当前直播间没开舰长。点开b站自带的选择列表三个也没有当前直播间的勋章，但是能手动换上

**Reply by the1812** [reply]
这个组件的选项里就可以控制是否自动佩戴勋章, 勋章大于 20 级的样式问题下个版本会修
